WordPress/gutenberg
# Fix focus loss when navigating the guide component

Extracted from #32765

## What?

For some reason, in the React 18 upgrade PR #32765, `useFocusOutside` works better than trunk (I don't know why yet). Which means there were some failing e2e tests that actually show real bugs we have on trunk, more precisely "focus loss" bugs. 

One of them is in the guide component (editor welcome guide): On trunk, on chrome, you can try clicking "previous" on the second page, you'll notice that the focus is lost (document.activeElement equal to body) but for some weird reason in trunk, the modal doesn't close.

This PR solves the focus loss in the Guide component.

## Why?

Focus loss is an a11y issue even if the modal is kept open right now.

## How?

I've noticed that there was already a behavior in place that auto-focuses the "finish" button. this PR removes that specific behavior and implements a generic solution that handle all cases:

 - If a focus loss happen, focus the "next button" if present otherwise, focus the "finish button", otherwise finds the first focusable element in the modal.

## Testing Instructions

1- Open the editor welcome guide
2- Click once on "next"
3- Click on "previous" to get back to the first page
4- Check the active element, it should be in the "next button".

## Screenshots or screencast
